manifests/mod/deflate.pp

```diff
@@ -12,6 +12,7 @@
     'Ratio'  => 'ratio'
   }
 ) {
+  include ::apache
   ::apache::mod { 'deflate': }
```

Comment:
Sorry, I forgot to say that I attached the files for profile::apache and
profile::apache::cfg::ontrack to the last email.

Damon

On Thu, Oct 29, 2015 at 3:14 PM, Damon Conway <vstraylight@gmail.com> wrote:

> I was confused as well. It did not error out there. The error I got is
> documented in the ticket.
>
> "Error: Could not retrieve catalog from remote server: Error 400 on
> SERVER: Invalid relationship: File[deflate.conf] { require => Exec[mkdir
> ] }, because Exec[mkdir ] doesn't seem to be in the catalog"
>
> Even though in my profile::apache I have "include ::apache" already, and
> my "profile::apache::cfg::ontrack" uses $::apache::httpd_dir just fine and
> is where I do a "include ::apache::mod::deflate".
>
> I came to the include ::apache solution from MODULES-1491.
>
> It seems as if puppet is not correctly resolving the catalog. The other
> solution I can think of is to include ::apache in
> profile::apache::cfg::ontrack and use anchors to make sure the apache class
> loads before apache::mod::deflate. Seems like a lot of work for the end
> user when this should just work.
>
> Thanks,
> Damon
>
>
> On Thu, Oct 29, 2015 at 8:45 AM, Igor Galić <notifications@github.com>
> wrote:
>
>> In manifests/mod/deflate.pp
>> <https://github.com/puppetlabs/puppetlabs-apache/pull/1236#discussion_r43402520>
>> :
>>
>> > @@ -12,6 +12,7 @@
>> >      'Ratio'  => 'ratio'
>> >    }
>> >  ) {
>> > +  include ::apache
>> >    ::apache::mod { 'deflate': }
>>
>> i'm confused.
>> shouldn't that line ^ error out, and tell you to first include apache
>> before including any modules?
>> see:
>> https://github.com/puppetlabs/puppetlabs-apache/blob/master/manifests/mod.pp#L11-L13
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/puppetlabs/puppetlabs-apache/pull/1236/files#r43402520>
>> .
>>
>
>